### PR TITLE
Simplify build and release

### PR DESCRIPTION
**Pull request by julien-truffaut**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,61 +1,49 @@
...
     - secure: lJxaR+9VM96s7rfaigL3K2FbcP3XiZ+AerbgOBOpYi2anRkMPQ91h1g60HYTwEDGJIKybyNWFuXZM8hNtQMCxzxgu7ev1SyRr5Otfzg5ehGqB5UWbSAgpXlqhR5dAonwmqyEEDF2OK+k1fa0N6mIwpQeuUCwrt68kTp4f0K9jMQ=
     - secure: NZQn8lB04I+FO3Om680CXWP29scS/zRhITbLO/1sM3dnuG8ygn0fzlFMQhgCO3fCjnclThAi+6MN7UnyV8BwDbFhCDe627+06px5fHUw2TPdzhz/BwngAW9pRsx6m0b8eBqQmt8Q6mlgeYOPemsFOeqQTKZulpIhSYnO1R1Nwzc=
   matrix:
     - TARGET=JVM PRE=test:compile
-    - TARGET=JS  PRE=test:fastOptJS
-sudo: true
-cache:
-  directories:
-    - $HOME/.ivy2/cache
-    - $HOME/.sbt/boot
```
Comment (dwijnand) [inline .travis.yml]: Keep these 4 lines, and make it `$HOME/.sbt` so you also cache the launchers.

File: build.sbt [hunk 2]
```diff
@@ -1,21 +1,49 @@
...
+      " 6b656e6a69@gmail.com",
+      url("https://github.com/xuwei-k")
+    ),
+  )
+))
 
 lazy val scalatestVersion = settingKey[String]("")
 
 lazy val buildSettings = Seq(
   organization       := "com.github.julien-truffaut",
```
Comment (dwijnand) [inline build.sbt]: This is redundant (set above)

File: build.sbt [hunk 3]
```diff
@@ -74,9 +86,7 @@ lazy val macroVersion = "2.1.1"
 lazy val paradisePlugin = Def.setting{
```
Comment (dwijnand) [inline build.sbt]: No longer necessary when only supporting 2.12+.

File: build.sbt [hunk 4]
```diff
@@ -87,14 +97,13 @@ lazy val paradisePlugin = Def.setting{
 lazy val kindProjector  = "org.typelevel"  % "kind-projector" % "0.10.0" cross CrossVersion.binary
 
 def mimaSettings(module: String): Seq[Setting[_]] = mimaDefaultSettings ++ Seq(
-  mimaPreviousArtifacts := Set("com.github.julien-truffaut" %  (s"monocle-${module}_2.11") % "1.3.0")
+  mimaPreviousArtifacts := Set("com.github.julien-truffaut" %  (s"monocle-${module}_2.12") % "1.6.0")
```
Comment (dwijnand) [inline build.sbt]: Use `%%`

**Comment by julien-truffaut** [comment]
> Nice spring cleaning!

Thanks Dale!

File: build.sbt (hunk 3 above)
Comment (julien-truffaut) [reply build.sbt]: I think it is still needed: https://travis-ci.org/julien-truffaut/Monocle/jobs/521607884

**Comment by julien-truffaut** [comment]
I don't understand why there are 5 jobs in travis: https://travis-ci.org/julien-truffaut/Monocle/builds/521621892?utm_source=github_status&utm_medium=notification

2.13..0-RC1 JVM test compile seems to be executed twice

**Comment by dwijnand** [comment]
Move the `script` for the test stage to top-level.  Then you're `include`-ing a release stage, redefining the script to run.

**Comment by julien-truffaut** [comment]
> Move the `script` for the test stage to top-level. Then you're `include`-ing a release stage, redefining the script to run.

Thanks for checking but I am not sure how can I reference the script for test stage if I move it to top level.

**Comment by dwijnand** [comment]
The matrix will multiple the script.  I'll send a PR.

**Comment by julien-truffaut** [comment]
Awesome, thanks very much Dale

**Pull request merged**